src/components/abstractComponent.ts

```diff
@@ -0,0 +1,579 @@
+import { ComponentContainer } from "./componentContainer";
+
+import { Point, SpaceRequest, Bounds } from "../core/interfaces";
+import * as RenderController from "../core/renderController";
+import * as Utils from "../utils";
+
+export type GenericComponentCallback<D> = (component: IComponent<D>) => void;
+
+export type IResizeHandler = (size: { height: number, width: number }) => void;
+
+export interface IContent {
+  node(): Node;
+}
+
+export interface IComponent<D> {
+  /**
+   * Attaches the Component as a child of a given html element.
+   *
+   * @param {D} selection.
+   * @returns {Component} The calling Component.
+   */
+  anchor(selection: D): this;
+  anchorHTML(selection: HTMLElement): this;
```

Comment:
this was a hack that allowed me to preserve back-comp for as long as possible while implementing. Let me see if I can remove this now and just officially change the anchoring behavior